core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -477,23 +484,27 @@ private[transaction] class TransactionMetadata(val transactionalId: String,
           }
 
         case PrepareAbort | PrepareCommit => // from endTxn
+          // In V2, we allow state transits from Empty, CompleteCommit and CompleteAbort to PrepareAbort. It is possible
+          // their updated start time is not equal to the current start time.
+          val allowedEmptyAbort = toState == PrepareAbort && transitMetadata.clientTransactionVersion.supportsEpochBump() &&
+            (state == Empty || state == CompleteCommit || state == CompleteAbort)
```

Comment:
Maybe create a Boolean flag here `validTimestamp` so that inside the condition we can check `!validTimestamp`?